src/main/java/htsjdk/variant/variantcontext/JEXLMap.java

```diff
@@ -11,152 +12,188 @@
+     * do we contain the specified key
+     * @param o the key
+     * @return true if we have a value for that key
      */
+    public boolean containsKey(Object o) { return jexl.containsKey(o); }
+
     public Set<JexlVCMatchExp> keySet() {
         return jexl.keySet();
     }
 
     /**
-     * get all the values of the map.  This is an expensive call, since it evaluates all keys that haven't
-     * been evaluated yet.  This is fine if you truely want all the keys, but if you only want a portion, or  know
+     * Get all the values of the map, i.e. the {@link Boolean} values.
+     * This is an expensive call, since it evaluates all keys that haven't been evaluated yet.
+     * This is fine if you truly want all the keys, but if you only want a portion, or  know
      * the keys you want, you would be better off using get() to get them by name.
+     *
+     * Note: due to laziness, this accessor actually modifies the instance by possibly forcing evaluation of an Jexl expression.
      * @return a collection of boolean values, representing the results of all the variants evaluated
+     *
+     * @throws IllegalArgumentException
      */
     public Collection<Boolean> values() {
-        // this is an expensive call
-        for (JexlVCMatchExp exp : jexl.keySet())
-            if (jexl.get(exp) == null)
+        for (final JexlVCMatchExp exp : jexl.keySet()) {
+            if (jexl.get(exp) == null) {
                 evaluateExpression(exp);
+            }
+        }
         return jexl.values();
     }
 
     /**
-     * evaulate a JexlVCMatchExp's expression, given the current context (and setup the context if it's null)
-     * @param exp the JexlVCMatchExp to evaluate
+     * TODO: the number may count invalid entries, is this good?
+     * @return the number of keys, i.e. {@link JexlVCMatchExp}'s hold by this mapping.
+     */
+    public int size() {
+        return jexl.size();
+    }
+
+    public boolean isEmpty() { return this.jexl.isEmpty(); }
+
+    // -----------------------------------------------------------------------------------------------
+    // Modifiers
+    // -----------------------------------------------------------------------------------------------
+
+    public Boolean put(JexlVCMatchExp jexlVCMatchExp, Boolean aBoolean) {
+        return jexl.put(jexlVCMatchExp, aBoolean);
+    }
+
+    public void putAll(Map<? extends JexlVCMatchExp, ? extends Boolean> map) {
+        jexl.putAll(map);
+    }
+
+    // -----------------------------------------------------------------------------------------------
+    // Utilities
+    // -----------------------------------------------------------------------------------------------
+
+    /**
+     * Lazily initializes {@link #jexl}, in the sense that all keys in {@code jexlCollection} are associated with {@code null}.
+     * @param jexlCollection
+     */
+    private void lazyInitialize(Collection<JexlVCMatchExp> jexlCollection) {
```

Comment:
I'm not sure this is any clearer than just initialize. Comment might read better as 
"Initializes all keys with null values indicating that they have not yet been evaluated.  The actual value will be computed only when the key is requested"